src/coreComponents/meshUtilities/CornerPointMeshGenerator.cpp

```diff
@@ -0,0 +1,191 @@
+
+CornerPointMeshGenerator::CornerPointMeshGenerator( string const & name, Group * const parent ):
+  MeshGeneratorBase( name, parent )
+{
+  registerWrapper( viewKeyStruct::filePathString(), &m_filePath ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setRestartFlags( RestartFlags::NO_WRITE ).
+    setDescription( "Path to the mesh file" );
+
+  string const builderName = "cpMeshBuilder";
+  m_cPMeshBuilder = std::make_unique< CPMesh::CPMeshBuilder >( builderName );
+}
+
+CornerPointMeshGenerator::~CornerPointMeshGenerator()
+{}
+
+void CornerPointMeshGenerator::postProcessInput()
+{
+  m_cPMeshBuilder->buildMesh( m_filePath );
+}
+
+Group * CornerPointMeshGenerator::createChild( string const & GEOSX_UNUSED_PARAM( childKey ),
+                                               string const & GEOSX_UNUSED_PARAM( childName ) )
+{
+  return nullptr;
+}
+
+void CornerPointMeshGenerator::generateMesh( DomainPartition & domain )
+{
+  Group & meshBodies = domain.getGroup( string( "MeshBodies" ));
+  MeshBody & meshBody = meshBodies.registerGroup< MeshBody >( this->getName() );
+
+  MeshLevel & meshLevel0 = meshBody.registerGroup< MeshLevel >( string( "Level0" ));
+  NodeManager & nodeManager = meshLevel0.getNodeManager();
+  CellBlockManager & cellBlockManager = domain.getGroup< CellBlockManager >( keys::cellManager );
+
+  // Step 0: transfer the neighbor list
+
+  // TODO: change the name, we don't use metis
+  domain.getMetisNeighborList() = m_cPMeshBuilder->neighborsList();
+
+  // we can start constructing the mesh
+
+  // Step 1: fill vertex information
+
+  arrayView2d< real64 const > vertices = m_cPMeshBuilder->vertices();
+  arrayView1d< globalIndex const > vertexToGlobalVertex = m_cPMeshBuilder->vertexToGlobalVertex();
+  localIndex const nVertices = vertices.size( 0 );
+  nodeManager.resize( nVertices );
+  arrayView1d< globalIndex > const & vertexLocalToGlobal = nodeManager.localToGlobalMap();
+  arrayView2d< real64, nodes::REFERENCE_POSITION_USD > const & X = nodeManager.referencePosition();
+
+  Group & vertexSets = nodeManager.sets();
+  SortedArray< localIndex > & allVertices =
+    vertexSets.registerWrapper< SortedArray< localIndex > >( string( "all" ) ).reference();
+
+  real64 xMin[3] = { std::numeric_limits< real64 >::max() };
+  real64 xMax[3] = { std::numeric_limits< real64 >::min() };
+
+  for( localIndex iVertex = 0; iVertex < nVertices; ++iVertex )
+  {
+    X( iVertex, 0 ) = vertices( iVertex, 0 );
+    X( iVertex, 1 ) = vertices( iVertex, 1 );
+    X( iVertex, 2 ) = vertices( iVertex, 2 );
+    allVertices.insert( iVertex );
+    vertexLocalToGlobal( iVertex ) = vertexToGlobalVertex( iVertex );
+
+    for( int dim = 0; dim < 3; dim++ )
+    {
+      if( X( iVertex, dim ) > xMax[dim] )
+      {
+        xMax[dim] = X( iVertex, dim );
+      }
+      if( X( iVertex, dim ) < xMin[dim] )
+      {
+        xMin[dim] = X( iVertex, dim );
+      }
+    }
+  }
+  LvArray::tensorOps::subtract< 3 >( xMax, xMin );
```

Comment:
I think that this whole loop to find some min and max could ce in its own function.